examples/cpp/route_guide/route_guide_callback_server.cc

```diff
@@ -228,25 +230,21 @@ class RouteGuideImpl final : public RouteGuide::CallbackService {
 
      private:
       void NextWrite() {
-        while (notes_iterator_ != received_notes_->end()) {
-          const RouteNote& n = *notes_iterator_;
+        if (notes_iterator_ != to_send_notes_.end()) {
+          StartWrite(&*notes_iterator_);
           notes_iterator_++;
-          if (n.location().latitude() == note_.location().latitude() &&
-              n.location().longitude() == note_.location().longitude()) {
-            StartWrite(&n);
-            return;
-          }
+        } else {
+          mu_->lock();
+          received_notes_->push_back(note_);
+          mu_->unlock();
+          StartRead(&note_);
         }
-        // Didn't write anything, so all done with this note
-        received_notes_->push_back(note_);
-        mu_->unlock();
-        StartRead(&note_);
       }
       RouteNote note_;
       std::mutex* mu_;
       std::vector<RouteNote>* received_notes_;
+      std::vector<RouteNote> to_send_notes_;
```

Comment:
let's add ABSL_GUARDED_BY notations

Reply:
(wherever applicable)

Reply:
Ah, we've polluted the examples with absl already. So that's doable, yes.

Reply:
Right so I had to also switch to using `absl::Mutex` instead of `std::mutex`, since the annotation wouldn't work otherwise.

Since it's only supposed to be an example, I think it's fine to use `absl::Mutex` directly without our typical abstraction layers around it.